common/sgx/verifier.c

```diff
@@ -101,6 +101,25 @@ static oe_result_t _verify_local_report(
 #endif
 }
 
+static oe_result_t _verify_claims_hash(
+    sgx_quote_t* quote,
+    const uint8_t* claims,
+    size_t claims_size)
+{
+    oe_result_t result = OE_UNEXPECTED;
+    OE_SHA256* hash = (OE_SHA256*)quote->report_body.report_data.field;
+    OE_SHA256 claims_hash;
+    oe_sha256_context_t hash_ctx = {0};
+
+    OE_CHECK(oe_sha256_init(&hash_ctx));
+    OE_CHECK(oe_sha256_update(&hash_ctx, claims, claims_size));
+    OE_CHECK(oe_sha256_final(&hash_ctx, &claims_hash));
+
+    result = memcmp(hash, &claims_hash, OE_SHA256_SIZE) ? OE_FAILURE : OE_OK;
```

Comment:
I'd suggest adding a new error code in oe_result_t to indicate the hash value mismatch. The failure most likely occurs when the claims attached to the evidence is modified.

Reply:
Yes, I agree.
Any one else suggest doing so? And what kind of name we should use?

Reply:
How about OE_VERIFY_CLAIMS_TAMPERED? I saw several error codes in forms of OE_VERIFY_XXX